kaitaistruct.py

```diff
@@ -335,30 +336,35 @@ def bytes_terminate(data, term, include_term):
 
     @staticmethod
     def process_xor_one(data, key):
+        if key == 0:
+            return data
+
         if PY2:
             return bytes(bytearray(v ^ key for v in bytearray(data)))
         else:
             return bytes(v ^ key for v in data)
 
     @staticmethod
     def process_xor_many(data, key):
+        if key == b'\x00' * len(key):
+            return data
+        if len(key) == 1:
+            return KaitaiStream.process_xor_one(data, ord(key))
+
         if PY2:
             return bytes(bytearray(a ^ b for a, b in zip(bytearray(data), itertools.cycle(bytearray(key)))))
         else:
             return bytes(a ^ b for a, b in zip(data, itertools.cycle(key)))
 
+    precomputed_rotations = {(amount,1):[(i << amount) & 0xff | (i >> (-amount & 7)) for i in range(256)] for amount in range(9)}
```

Comment:
Not sure that memory lookup (especially in interpreted fully object-oriented language, so *I guess* CPU cache helps little) is faster than bit shift. Could you provide some benchmark results?

Reply:
I am unable to run a benchmark.

Reply:
I also need to withdraw one of the commits.

Reply:
>I am unable to run a benchmark.

there is no need to run kaitai benchmark, just an own benchmark